course_util.py

```diff
@@ -31,157 +31,187 @@
-        else:
-            test_list.append(response)
-            #writefile(str(request_id), str(response))
-        itersubmissions(response)
-        pass
-
-batch_c = classroom.new_batch_http_request(callback=callback_c)
-batch_cw = classroom.new_batch_http_request(callback=callback_cw)
-batch_s = classroom.new_batch_http_request(callback=callback_s)
-
-def user_courses():
-    #print('USER COURSES:\n')
-    batch_c.add(classroom.courses().list(fields='courses/name,courses/id'))
-
-
-def course_teachers(courseId):
-    #print('COURSE TEACHERS:\n')
-    batch_cw.add(classroom.courses().teachers().list(courseId=courseId, fields='teachers/courseId,teachers/userId,teachers/profile'))
-
+COURSE_FIELDS = ''
+TEACHER_FIELDS = ['teachers/courseId','teachers/userId','teachers/profile']
+STUDENT_FIELDS = ['students/courseId','students/userId','students/profile']
+COURSEWORK_FIELDS = ['courseWork/id','courseWork/title','courseWork/maxPoints','courseWork/description','courseWork/creationTime','courseWork/alternateLink','courseWork/courseId']
+SUBMISSION_FIELDS = ['studentSubmissions/id','studentSubmissions/assignedGrade','studentSubmissions/alternateLink','studentSubmissions/courseWorkId','studentSubmissions/courseId','studentSubmissions/userId','studentSubmissions/state']
 
-def course_students(courseId):
-    #print('COURSE STUDENTS:\n')
-    batch_cw.add(classroom.courses().students().list(courseId=courseId, fields='students/courseId,students/userId,students/profile'))
 
+def writefile(name, content):
+    save_path = os.path.abspath("logs")
+    completename = os.path.join(save_path, str(name))
+    with open(completename+".json", "w") as f:
+        f.write(content)
 
-def course_coursework(courseId):
-    #print('COURSE COURSEWORK\n')
-    batch_cw.add(classroom.courses().courseWork().list(courseId=courseId, fields='courseWork/id,courseWork/title,courseWork/maxPoints,courseWork/description,courseWork/creationTime,courseWork/alternateLink,courseWork/courseId'))
+def write_submissions(submissions, filename_suffix=''):
+    # Save submission JSON blob to `submissions.json`
+    string = json.dumps(submissions, sort_keys=True, indent=4)
+    writefile('submissions' + filename_suffix, string)
+    csv.fullconvert()
 
+def log_all(results):
+    for res, l in enumerate(results):
+        GCLOGGER.info(res)
 
-def coursework_submissions(courseId, courseworkId):
-    #print('COURSEWORK SUBMISSIONS\n')
-    batch_s.add(classroom.courses().courseWork().studentSubmissions().list(courseId=courseId, courseWorkId=courseworkId, fields='studentSubmissions/id,studentSubmissions/assignedGrade,studentSubmissions/alternateLink,studentSubmissions/courseWorkId,studentSubmissions/courseId,studentSubmissions/userId,studentSubmissions/state'))
+def format_all(results):
+    return course_utiljs.formatAll(results)
 
 
-def iterteachers(resp):
-    #print('TEACHERS:\n')
-    for i, t in enumerate(resp['courses']):
-        course_teachers(t['id'])
+class CourseFetcher:
```

Comment:
Move this into a class so we can have a classroom service per instance instead of using a global variable.